redirects.go

```diff
@@ -94,8 +98,16 @@ func Must(v []Rule, err error) []Rule {
 
 // Parse the given reader.
 func Parse(r io.Reader) (rules []Rule, err error) {
-	s := bufio.NewScanner(r)
+	// not too large
+	b, err := bufio.NewReaderSize(r, maxFileSizeInBytes+1).Peek(maxFileSizeInBytes + 1)
+	if err != nil && err != io.EOF {
+		return nil, err
+	}
+	if len(b) > maxFileSizeInBytes {
+		return nil, fmt.Errorf("redirects file size cannot exceed %d bytes", maxFileSizeInBytes)
+	}
 
+	s := bufio.NewScanner(bytes.NewReader(b))
 	for s.Scan() {
```

Comment:
@Jorropo any room for improvement here? I guess we need to read entire thing anyway, so does not matter much, bu just to be sure.

Reply:
I don't see why you use `bufio`, it's a complex beast inside.

You can just do `io.ReadAll(&io.LimitedReader{r, maxFileSizeInBytes+1})`.

Reply:
Thanks, I've switched to simpler code with LimitedReader and streaming in https://github.com/ipfs/go-ipfs-redirects-file/pull/18